test/e2e/cli/kubecli/kubecli.go

```diff
@@ -1,37 +1,60 @@
 package kube
 
 import (
-	"encoding/json"
+	"context"
 	"encoding/pem"
 	"fmt"
 	"os"
-	"strings"
 
-	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/connectionsecret"
+	"gopkg.in/yaml.v2"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
 
 	. "github.com/onsi/gomega"
 	. "github.com/onsi/gomega/gbytes"
 	"github.com/onsi/gomega/gexec"
-
 	"github.com/sethvargo/go-password/password"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 
 	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/connectionsecret"
 	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli"
 	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/utils"
 )
 
-// GenKubeVersion
-func GenKubeVersion(fullVersion string) string {
-	version := strings.Split(fullVersion, ".")
-	return fmt.Sprintf("Major:\"%s\", Minor:\"%s\"", version[0], version[1])
+func CreateNewClient() (client.Client, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	err = v1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return nil, err
+	}
+	k8sClient, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		return nil, err
+	}
+	return k8sClient, nil
 }
 
 // GetPodStatus status.phase
-func GetPodStatus(ns string) func() string {
-	return func() string {
-		session := cli.Execute("kubectl", "get", "pods", "-l", "app.kubernetes.io/instance=mongodb-atlas-kubernetes-operator", "-o", "jsonpath={.items[0].status.phase}", "-n", ns)
-		return string(session.Wait("1m").Out.Contents())
+func GetPodStatus(ctx context.Context, k8sClient client.Client, ns string) (string, error) {
+	pod := &corev1.PodList{}
+	err := k8sClient.List(ctx, pod, client.InNamespace(ns), client.MatchingLabels{"app.kubernetes.io/instance": "mongodb-atlas-kubernetes-operator"})
```

Comment:
Here `k8sClient` is clearly being used so we should update the directory names and change references.

Reply:
Good point. Done